packages/backend/server/src/__tests__/copilot.e2e.ts

```diff
@@ -726,3 +736,67 @@ test('should be able to search image from unsplash', async t => {
   const resp = await unsplashSearch(app, token);
   t.not(resp.status, 404, 'route should be exists');
 });
+
+test('should be able to manage context', async t => {
+  const { app, context } = t.context;
+
+  const { id: workspaceId } = await createWorkspace(app, token);
+  const sessionId = await createCopilotSession(
+    app,
+    token,
+    workspaceId,
+    randomUUID(),
+    promptName
+  );
+
+  // use mocked embedding client
+  Sinon.stub(context, 'embeddingClient').get(() => new MockEmbeddingClient());
+
+  {
+    await t.throwsAsync(
+      createCopilotContext(app, token, workspaceId, randomUUID()),
+      { instanceOf: Error },
+      'should throw error if create context with invalid session id'
+    );
+
+    const context = createCopilotContext(app, token, workspaceId, sessionId);
+    await t.notThrowsAsync(context, 'should create context with chat session');
+
+    const list = await listContext(app, token, workspaceId, sessionId);
+    t.deepEqual(list, [{ id: await context }], 'should list context');
+  }
+
+  const fs = await import('node:fs');
+  const buffer = fs.readFileSync(
+    new URL('../../../../common/native/fixtures/sample.pdf', import.meta.url)
+  );
+
+  {
+    const contextId = await createCopilotContext(
+      app,
+      token,
+      workspaceId,
+      sessionId
+    );
+
+    const fileId = await addContextFile(
+      app,
+      token,
+      contextId,
+      randomUUID(),
+      'sample.pdf',
+      buffer
+    );
+    const [file] =
+      (await listContextFiles(app, token, workspaceId, sessionId, contextId)) ||
+      [];
+    t.is(file.id, fileId, 'should list file');
+    t.is(file.status, 'finished', 'should list file status');
+    t.is(file.name, 'sample.pdf', 'should list file name');
+    t.is(file.chunk_size, 3, 'should split file into chunks');
+
+    const result = (await matchContext(app, token, contextId, 'test', 2))!;
```

Comment:
It looks like `sample.pdf` does not contain anything similar to `test`. Why this `matchContext` function work?

Reply:
mock